src/GovUk.Education.ExploreEducationStatistics.Public.Data.Api.Tests/Controllers/DataSetsControllerTests.cs

```diff
@@ -714,440 +719,859 @@ public GetDataSetMetaTests(TestApplicationFactory testApp) : base(testApp)
+
+                Assert.Equal(2, validationProblem.Errors.Count);
+
+                var errorDetail1 = GetErrorDetail(validationProblem.Errors[0]);
+                var errorDetail2 = GetErrorDetail(validationProblem.Errors[1]);
+
+                Assert.NotNull(errorDetail1);
+                Assert.NotNull(errorDetail2);
+                Assert.Equal(2, errorDetail1.Count);
+                Assert.Equal(2, errorDetail2.Count);
+                Assert.Equal("invalid1", errorDetail1["value"].GetString());
+                Assert.Equal("invalid2", errorDetail2["value"].GetString());
+                Assert.Equal(
+                    EnumUtil.GetEnums<MetadataType>(),
+                    errorDetail1["allowed"].EnumerateArray().Select(e => EnumUtil.GetFromEnumValue<MetadataType>(e.GetString()!)));
+                Assert.Equal(
+                    EnumUtil.GetEnums<MetadataType>(),
+                    errorDetail2["allowed"].EnumerateArray().Select(e => EnumUtil.GetFromEnumValue<MetadataType>(e.GetString()!)));
+            }
+
+            [Fact]
+            public async Task MixedValidAndInvalidMetadataTypes_AllowedValueError()
+            {
+                var response = await GetDataSetMeta(Guid.NewGuid(), types: [MetadataType.Filters.ToString(), "invalid"]);
+
+                var validationProblem = response.AssertValidationProblem();
+
+                validationProblem.AssertHasAllowedValueError("types[1]");
+
+                var error = Assert.Single(validationProblem.Errors);
+
+                var errorDetail = GetErrorDetail(error);
+
+                Assert.NotNull(errorDetail);
+                Assert.Equal(2, errorDetail.Count);
+                Assert.Equal("invalid", errorDetail["value"].GetString());
+                Assert.Equal(
+                    EnumUtil.GetEnums<MetadataType>(),
+                    errorDetail["allowed"].EnumerateArray().Select(e => EnumUtil.GetFromEnumValue<MetadataType>(e.GetString()!)));
+            }
+
+            private static Dictionary<string, JsonElement> GetErrorDetail(ErrorViewModel error)
+            {
+                var errorDetailJson = Assert.IsType<JsonElement>(error.Detail);
+                var errorDetail = errorDetailJson.Deserialize<Dictionary<string, JsonElement>>();
+
+                Assert.NotNull(errorDetail);
+                return errorDetail;
+            }
         }
 
-        private async Task<HttpResponseMessage> GetDataSetMeta(Guid dataSetId, string? dataSetVersion = null)
+        private async Task<HttpResponseMessage> GetDataSetMeta(
+            Guid dataSetId,
+            string? dataSetVersion = null,
+            IReadOnlyList<string>? types = null)
         {
             var query = new Dictionary<string, string?>
             {
                 { "dataSetVersion", dataSetVersion?.ToString() },
             };
 
+            if (types is not null)
+            {
+                query.Add("types", types.JoinToString(","));
+            }
```

Comment:
We might want to do another test case where we check that array query parameter syntax is valid as well e.g. `?types[0]=Filters&types[1]=Locations`